cmd/ebrelayer/siflogger/siflogger.go

```diff
@@ -0,0 +1,125 @@
+package siflogger
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/go-kit/kit/log/term"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+type Level byte
+
+const (
+	Debug Level = iota
+	Info
+	Error
+	None
+)
+
+type SifLogger struct {
+	logger log.Logger
+}
+
+func colorFn(keyvals ...interface{}) term.FgBgColor {
+	if keyvals[0] != level.Key() {
+		panic(fmt.Sprintf("expected level key to be first, got %v", keyvals[0]))
+	}
+
+	switch keyvals[1].(level.Value).String() {
+	case "debug":
+		return term.FgBgColor{Fg: term.DarkGreen}
+	case "error":
+		return term.FgBgColor{Fg: term.Red}
+	default:
+		return term.FgBgColor{}
+	}
+}
+
+var filterNumber = 0
+
+func LogFileLine(depth int) kitlog.Valuer {
+	return func() interface{} {
+		_, file, line, _ := runtime.Caller(depth + filterNumber)
+		idx := strings.LastIndexByte(file, '/')
+		return file[idx+1:] + ":" + strconv.Itoa(line)
+	}
+}
+
+func New() SifLogger {
+	logger := log.NewTMLoggerWithColorFn(log.NewSyncWriter(os.Stdout), colorFn)
+	logger = logger.With("caller", LogFileLine(5))
+	e := SifLogger{logger}
+	filterNumber = 0 // supposedly used as singleton
+	return e
+}
+
+func (e *SifLogger) SetGlobalFilter(level Level) {
+	var option log.Option
+	switch level {
+	case Debug:
+		option = log.AllowDebug()
+	case Error:
+		option = log.AllowError()
+	case Info:
+		option = log.AllowInfo()
+	case None:
+		option = log.AllowNone()
+	default:
+		panic("incorrect level")
+	}
+	filterNumber++
+	filter := log.NewFilter(e.logger, option)
+	e.logger = filter
+}
+
+func (e *SifLogger) SetFilterForLayer(level Level, keyvals ...interface{}) {
```

Comment:
Obviously the keyvals' data structure is map. Don't need judge the length is even if input is a map.

Reply:
it is my question for the team as well. Do we really need a map? My opinion and idea was to have only keys without values (I call them "tags"), because when you want isolate something in logging you use only one word. I don't understand the case where it could be for example Tag=1 and somewhere else Tag=2. Logically speaking, if you have several and many values for Tag then for what reason you wanted to split your code to Tag itself? Not sure I understand fully logging concept of tendermint. BTW they have reserved key "module" hardcoded in tendermint internals, so from this point of view is possible to have "module"="1" and "module"="2". But other than that I think simple chain of tags is enough like in example in the current code (look at goroutines): 
logger = root.Tag("relayer")
logger_eth = logger.Tag("eth")
logger_cosmos = logger.Tag("cosmos")
logger_tendermint_client = logger_cosmos.Tag("client")
so we will be able to filter by "eth" OR by "cosmos" including all embedded. And additionally filter by tenderming client inside "cosmos" ONLY.

Reply:
If you think the value not useful, just remove it. the input as key/tag list is fine.